### PR TITLE
Update dependencies; add CI checks

### PR DESCRIPTION
**Pull request by cqc-alec**
There is a strange [test failure on MacOS](https://github.com/CQCL/pytket-pennylane/issues/2) which is skipped here; this needs further investigation.

We should be able to put `QulacsBackend` back in the tests when pytket-qulacs has been [updated](https://github.com/CQCL/pytket-extensions/issues/478) to use the latest version.

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+API documentation
+~~~~~~~~~~~~~~~~~
+
+.. automodule:: pytket.extensions.pennylane
```
Comment (ss2165) [inline docs/api.rst:4]: Does this include the `pennylane_to_tk` function?

**Review by ss2165 — COMMENTED** [review]
Thanks Alec, just a question about the docs.

File: docs/api.rst (hunk 1 above)
Comment (cqc-alec) [reply docs/api.rst:4]: No it doesn't, I wasn't sure if it was required to be public (or just used by `PytketDevice.apply()`), but can easily add it.

Comment (ss2165) [reply docs/api.rst:4]: yes please, sorry there should have been a docstring on there

**Review by ss2165 — APPROVED** [review]
👍

**Pull request merged**